src/rgw/rgw_op.cc

```diff
@@ -3057,10 +3048,10 @@ void RGWCreateBucket::execute()
   buffer::list aclbl;
   buffer::list corsbl;
   bool existed;
-  string bucket_name;
-  rgw_make_bucket_entry_name(s->bucket_tenant, s->bucket_name, bucket_name);
+  string bucket_name = rgw_make_bucket_entry_name(s->bucket_tenant, s->bucket_name);
   rgw_raw_obj obj(store->svc()->zone->get_zone_params().domain_root, bucket_name);
   obj_version objv, *pobjv = NULL;
+  std::unique_ptr<rgw::sal::RGWUser> user(new rgw::sal::RGWRadosUser(store, *s->user));
```

Comment:
why unique_ptr (why why heap-allocate) here?

Reply:
RGWUser is a pure virtual, so it can't be instantiated.  The unique_ptr makes it go away at the end of the function call.  This is all temporary, until the zipper objects are integrated into req_state, so it doesn't matter that much how it's done.

Reply:
presumably you could put `rgw::sal::RGWRadosUser user` on the stack here

but my recollection is that these handles eventually need to be reference counted to support things like caching layers, which does imply heap allocation